src/core/Akka.Remote/RemoteActorRefProvider.cs

```diff
@@ -532,30 +532,27 @@ public RemoteDeadLetterActorRef(IActorRefProvider provider, ActorPath actorPath,
 
             protected override void TellInternal(object message, IActorRef sender)
             {
-                message
-                    .Match()
-                    .With<EndpointManager.Send>(
-                        send =>
-                        {
-                            // else ignore: it is a reliably delivered message that might be retried later, and it has not yet deserved
-                            // the dead letter status
-                            if (send.Seq == null)
-                            {
-                                base.TellInternal(send.Message, send.SenderOption ?? ActorRefs.NoSender);
-                            }
-                        })
-                    .With<DeadLetter>(
-                        deadLetter =>
-                        {
-                            // else ignore: it is a reliably delivered message that might be retried later, and it has not yet deserved
-                            // the dead letter status
-                            var deadSend = deadLetter.Message as EndpointManager.Send;
-                            if (deadSend != null && deadSend.Seq == null)
-                            {
-                                base.TellInternal(deadSend.Message, deadSend.SenderOption ?? ActorRefs.NoSender);
-                            }
-                        })
-                    .Default(_ => base.TellInternal(message, sender));
+                var send = message as EndpointManager.Send;
+                var deadLetter = message as DeadLetter;
+                if (send != null)
+                {
+                    if (send.Seq == null)
+                    {
+                        base.TellInternal(send.Message, send.SenderOption ?? ActorRefs.NoSender);
+                    }
+                }
+                else if (deadLetter?.Message is EndpointManager.Send)
```

Comment:
I've replaced the pattern matching code with normal if blocks, this specific check only consumes deadletters containing EndpointManager.Send messages now.
(unlike before where it incorrectly consumed all deadletters)